### ISSUE TITLE
`sh:icon` property

### ISSUE THREAD

**Issue by jeswr** [w3c/data-shapes#207]
Similarly to how there is reason to have the `sh:name` and `sh:description` properties for `sh:NodeShape`s and `sh:PropertyShape`s, I believe that it would be appropriate to have a `sh:icon` property to support displaying shapes to users / developers.

In particular; the use-case I have encountered is the creation of a catalogue of shapes for application developers, where icons could be used in displaying the shape to developers browsing shapes they want to re-use in developing applications.

Since the SHACL 1.2 work is coming up, I thought it might be an appropriate time to try and get such a property added to the spec.

**Comment by tfrancart** [comment]
I had the same need of associating icons to (node) shapes for the [Sparnatural query UI ](https://sparnatural.eu/).
However in a broader linked data perspective you may need to associate an icon to virtually any resource, be it an OWL class, a property, an instance, etc. I don't think this is specific to SHACL. This is why I created a small vocabulary "[volipi](http://data.sparna.fr/ontologies/volipi/index-en.html)" (VOcabulary for LInked data Presentation) containing [volipi:icon](http://data.sparna.fr/ontologies/volipi#icon) to associate an icon file to a resource, [volipi:iconName](http://data.sparna.fr/ontologies/volipi#iconName) to associate an icon name from an icon library e.g. FontAwesome, or [volipi:class](http://data.sparna.fr/ontologies/volipi#class) to associate a CSS class to a resource.

**Comment by HolgerKnublauch** [comment]
Yes the requirement for icons exists, and for all resources. We are supporting custom icons in TopBraid using a slight indirection, see https://datashapes.org/propertyroles.html#IconRole

This means that any ontology can declare its own "icon" properties and just mark them as dash:propertyRole dash:IconRole. As shown in the link above, this means that:
- icons can be directly asserted as individual triples for each subject
- icons can be set as sh:defaultValue
- icons can be inferred using SHACL rules

If the new SHACL WG produces official SHACL inferencing support then the latter means for example that classes marked as owl:deprecated can display with a different icon than classes that are not, or cities that are capitals can display differently on a map than normal cities, or cities with more than 1 million inhabitants can render with a larger icon etc.

**Comment by jeswr** [comment]
Without being up to date on the work on inferencing support -  I guess the question is whether the `dash:propertyRole`; and the built-in property roles in https://datashapes.org/propertyroles.html#PropertyRoles should be ported into the SHACL namespace.

I'm inclined to say yes - if for no other reason than we can look into adding some shorthand to SHACL-C for this feature.

Happy to discuss this in the next call.
